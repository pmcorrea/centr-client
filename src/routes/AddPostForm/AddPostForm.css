.AddPost__form_container {
	flex-direction: column;
	width: 100vw;
  	align-items: center;
  	justify-content: center;
}

.AddPost__form {
	background-color: white;
	width: 400px;
	margin: 20px auto;
	width: 80vw;
  }

.AddPost__input,
.AddPost__select, 
.AddPost__option {
	font-size: 15px;
	padding: 8px;
	width: 100%;
	border: none;
  border-radius: 0px;
  /* margin-top: 10px; */
  margin-bottom: 30px;
  }

.AddPost__textarea {
  width: 100%;
  height: 150px;
  resize: none;
  margin-top: 10px;
  font-size: 15px;
  border-radius: 10px;
  border: 2px rgba(214, 214, 214, 0.63) solid;
  padding: 10px;
  margin-bottom: 10px;
  }

  #folder_select_menu_label {
    color: gray;
  }

  .AddPost__select{
    border-bottom: 2px rgba(214, 214, 214, 0.63) solid;
    border-radius: 2px;
  }


  .AddPost__textarea:hover{
    border-color: orange;
  }

  .AddPost__select:hover{
    border-color: orange;
  }


  .AddPost__input {
	border-top: 0px;
  border-right: 0px;
  border-left: 0px;
  border-bottom: 2px rgba(214, 214, 214, 0.63) solid;
  width: 100%; 
  /* margin-top: 10px; */
  font-size: 15px;
  color: black;
  background-color: rgb(255, 255, 255);
  }

  .AddPost__input:hover{
    border-color: orange;
  }


  .AddPost__input::placeholder {
    color: gray;
    font-size: 15px;
  }

  /* .AddPost__button {
	padding: 8px;
  border-radius: 0px;
  margin-top: 10px;
	width: 100%; 
	padding: 10px;
	font-size: 20px;
	background-color: black;
	color: white;
	border-radius: 0px;
	border: none;
	margin-top: 10px; 
	font-weight: 400;
  } */

.AddPost__button {
	font-size: 15px;
	padding: 15px;
	width: 100%;
	border: 0px solid white;
  border-radius: 5px;
  margin-top: 10px;
  font-weight: bold;
  background-color: rgb(112, 39, 161);  
  color: white;
  box-shadow: 0px 5px 20px rgba(128, 128, 128, 0.733);
}

.AddPost__button:hover{
  color: orange;
}
  
p {
	color: white;
}