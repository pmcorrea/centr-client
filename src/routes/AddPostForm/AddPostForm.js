import React, { Component } from 'react'
import './AddPostForm.css'
import MainContext from '../../contexts/MainContext';
import AuthApiService from "../../services/auth-api-service.js"


export default class AddPostForm extends Component {
  static contextType = MainContext;

  constructor(props) {
  super(props);
  this.state = {
		folder_id: null,
		post_name: '',
		modified: this.makeDate(),
		content: '',
		touched: false,
		error: null,
		visibility: 'Private',
		user_name: null,
		avatar: null
  }
}

componentDidMount() {
    this.setCurrentUser()
  }

  setCurrentUser() {
	AuthApiService.getCurrentUserDetailsById()
	.then(currentUser => {
		let avatar = currentUser[0]['avatar']
		currentUser = currentUser[0]

		this.setState({
			user_name: currentUser,
			avatar: avatar
		})
	})
	.catch(error => {
		this.setState({ error }, () => console.log(this.state.error))
	})
}
	makeDate() {
		let date = new Date()
		return date.toLocaleString()
	}

	updatePostVisibility(input) {
		this.setState({
			visibility: input,
		});
	}

  updatePostName(input) {
	  this.setState({
		  post_name: input,
	  });
  }

  updateFolderId(input) {
	  if (!this.state.touched) {
		this.setState({
			folder_id: input,
			touched: true
		});
	  }
	
}

  updatePostContent(input) {
		this.setState({
			content: input
		});
	}
	 
  updatePostFolder(input, folders) {
	  const singleFolder = folders.find(folder => folder.folder_name === input)
	  this.setState({
		folder_id: singleFolder.id, 
	  });
  }

  handleSubmit(event) {
	  event.preventDefault();
	  const post = this.state;

		AuthApiService.postPost(post)
			.then(result => {
			this.context.getDataWithToken()
			this.props.history.goBack()
		})
		.catch(error => 
			error => this.setState({error})
		) 

}

	validateSubmission() {
		const postName = this.state.post_name
		const postContent = this.state.content

		if (
				((postName.trim().length < 3) ||
				(postName.length < 3)
				) 
				||
				(postContent.length < 3)
			)
			{
			return true
		}
	}

	randomKey() {
		return Math.random().toString(36).substr(2, 9);
	}

  render() {
	const { error } = this.state;
	const {folders=[]} = this.context;
	let firstFolder = folders[0]
	let firstFolderId;
	  
	if (firstFolder) {
		firstFolderId = firstFolder['id']
	}
	
	return folders 
		? (<>
			<div role="alert">{error && <p>{error}</p>}</div>
			
			<div className="AddPost__form_contaner">
				<form onSubmit={e => this.handleSubmit(e)} className="AddPost__form">
					{/* <label htmlFor="post_input_field" id="folder_select_menu_label">post title</label> */}
					<input className="AddPost__input" placeholder="post title" 
						id='post_input_field' type='text' onChange={e => {
							this.updatePostName(e.target.value)
							this.updateFolderId(firstFolderId)
						}}></input>

					<label htmlFor="folder_select_menu" id="folder_select_menu_label">select folder</label>
					<select className="AddPost__select" id='folder_select_menu' onChange={e => this.updatePostFolder(e.target.value, folders)}>
						{folders.map(folder => (
							<option className="AddPost__option" value={folder.folder_name} 
								key={`folder.folder_name_${this.randomKey()}`}>{folder.folder_name}</option>
						))}
					</select>

					<label htmlFor="post_visibility_option" id="folder_select_menu_label">select visibility</label>
					<select className="AddPost__select" id='post_visibility_option' onChange={e => this.updatePostVisibility(e.target.value, folders)}>
							<option className="AddPost__option" value='Private' key={'Private'}>Private</option>
							<option className="AddPost__option" value='Public' key={'Public'}>Public</option>
					</select>

					<textarea className="AddPost__textarea" placeholder="add some content..." 
						id='content_area' type='text' 
						onChange={e => this.updatePostContent(e.target.value)}></textarea>
						
					<button className="AddPost__button" type='submit' 
						disabled={this.validateSubmission() }>submit</button>
				</form>
			</div>
		</>) 
		: ('');
	}
}