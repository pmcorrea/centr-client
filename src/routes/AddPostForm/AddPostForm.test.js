import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddPostForm from './AddPostForm'

describe(`AddPostForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.AddPostForm by default', () => {
    const wrapper = shallow(<AddPostForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the AddPostForm given props', () => {
    const wrapper = shallow(<AddPostForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
