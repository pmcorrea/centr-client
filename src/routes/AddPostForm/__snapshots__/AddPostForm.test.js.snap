// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPostForm component renders a form.AddPostForm by default 1`] = `
<Fragment>
  <div
    role="alert"
  />
  <div
    className="AddPost__form_contaner"
  >
    <form
      className="AddPost__form"
      onSubmit={[Function]}
    >
      <button
        className="go_back_button"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="arrow-alt-circle-left"
          fixedWidth={false}
          flip={null}
          icon="arrow-alt-circle-left"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="3x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <input
        className="AddPost__input"
        id="post_input_field"
        onChange={[Function]}
        placeholder="post title"
        type="text"
      />
      <label
        htmlFor="folder_select_menu"
        id="folder_select_menu_label"
      >
        select folder
      </label>
      <select
        className="AddPost__select"
        id="folder_select_menu"
        onChange={[Function]}
      />
      <label
        htmlFor="post_visibility_option"
        id="folder_select_menu_label"
      >
        select visibility
      </label>
      <select
        className="AddPost__select"
        id="post_visibility_option"
        onChange={[Function]}
      >
        <option
          className="AddPost__option"
          key="Private"
          value="Private"
        >
          Private
        </option>
        <option
          className="AddPost__option"
          key="Public"
          value="Public"
        >
          Public
        </option>
      </select>
      <textarea
        className="AddPost__textarea"
        id="content_area"
        onChange={[Function]}
        placeholder="add some content..."
        type="text"
      />
      <button
        className="AddPost__button"
        disabled={true}
        type="submit"
      >
        submit
      </button>
    </form>
  </div>
</Fragment>
`;

exports[`AddPostForm component renders the AddPostForm given props 1`] = `
<Fragment>
  <div
    role="alert"
  />
  <div
    className="AddPost__form_contaner"
  >
    <form
      className="AddPost__form"
      onSubmit={[Function]}
    >
      <button
        className="go_back_button"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="arrow-alt-circle-left"
          fixedWidth={false}
          flip={null}
          icon="arrow-alt-circle-left"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="3x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <input
        className="AddPost__input"
        id="post_input_field"
        onChange={[Function]}
        placeholder="post title"
        type="text"
      />
      <label
        htmlFor="folder_select_menu"
        id="folder_select_menu_label"
      >
        select folder
      </label>
      <select
        className="AddPost__select"
        id="folder_select_menu"
        onChange={[Function]}
      />
      <label
        htmlFor="post_visibility_option"
        id="folder_select_menu_label"
      >
        select visibility
      </label>
      <select
        className="AddPost__select"
        id="post_visibility_option"
        onChange={[Function]}
      >
        <option
          className="AddPost__option"
          key="Private"
          value="Private"
        >
          Private
        </option>
        <option
          className="AddPost__option"
          key="Public"
          value="Public"
        >
          Public
        </option>
      </select>
      <textarea
        className="AddPost__textarea"
        id="content_area"
        onChange={[Function]}
        placeholder="add some content..."
        type="text"
      />
      <button
        className="AddPost__button"
        disabled={true}
        type="submit"
      >
        submit
      </button>
    </form>
  </div>
</Fragment>
`;
